kotlin/goodmetrics/src/main/kotlin/goodmetrics/MetricsFactory.kt

```diff
@@ -103,6 +105,9 @@ class MetricsFactory(
             metrics.timestampNanos = timeSource.epochNanos()
         }
         val duration = System.nanoTime() - metrics.startNanoTime
+        if (metrics.metricsBehavior == MetricsBehavior.NO_TOTALTIME) {
+            return
+        }
```

Comment:
should move this line 107 below this guard statement
```suggestion
        if (metrics.metricsBehavior == MetricsBehavior.NO_TOTALTIME) {
            return
        }
        val duration = System.nanoTime() - metrics.startNanoTime
```